### ISSUE TITLE
Sometimes receive invalid peer certificate: BadSignature in certain environments

### ISSUE THREAD

**Issue by westonpace** [lancedb/lancedb#718]
This is probably a dependency issue.  However, users are encountering this and we either need to figure out the root cause and fix it (via updating dependencies, changing configuration) or provide guidance on how to work around it.

**Comment by westonpace** [comment]
Current Status:

I'm able to reproduce this by using any aarch64 linux instance.
The source of the error appears to be rustls (or possibly ring) failing to verify the AWS server's TLS signature
The TLS signature is valid 
Building lance from source on the system makes the error go away

My best guess is the error is originating somehow from the way we are cross-compiling for aarch64.  However, I manually built rultls, cross compiled on my x86_64 ubuntu system, with the `ghcr.io/rust-cross/manylinux2014-cross:aarch64` docker container that maturin is using, and that appears to work correctly (I did need to set the `CFLAGS_aarch64_unknown_linux_gnu` variable like we do in GHA) so I'm not certain.

I'm going to see if I can build a wheel with the 2_28 manylinux image and see if the same error happens.
